circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -260,3 +267,105 @@ def decompose_observables(
     }
 
     return subobservables_by_subsystem
+
+
+def find_cuts(
+    circuit: QuantumCircuit,
+    optimization: dict[str, str | int],
+    constraints: dict[str, int],
+) -> tuple[QuantumCircuit, dict[str, Any]]:
+    """
+    Find cut locations in a circuit, given optimization settings and QPU constraints.
+
+    Args:
+        circuit: The circuit to cut
+        optimization: Settings for controlling optimizer behavior. Currently,
+            only a best-first optimizer is supported.
+        constraints: QPU constraints used to generate the cut location search space.
+
+    Returns:
+        A circuit containing :class:`.BaseQPDGate` instances. The subcircuits
+        resulting from cutting these gates will be runnable on the devices
+        specified in ``constraints``.
+
+        A metadata dictionary:
+            cuts: A list of length-2 tuples describing each cut in the output circuit.
+                The tuples are formatted as ``(cut_type: str, cut_id: int)``. The
+                cut ID is the index of the cut gate or wire in the output circuit's
+                ``data`` field.
+            sampling_overhead: The sampling overhead incurred from cutting the specified
+                gates and wires.
+    """
+    circuit_cco = qc_to_cco_circuit(circuit)
+    interface = SimpleGateList(circuit_cco)
+
+    opt_settings = OptimizationSettings.from_dict(optimization)
+
+    # Hard-code the optimization type to best-first
+    opt_settings.set_engine_selection("CutOptimization", "BestFirst")
+
+    constraint_settings = DeviceConstraints.from_dict(constraints)
+
+    # Hard-code the optimizer to an LO-only optimizer
+    optimizer = LOCutsOptimizer(interface, opt_settings, constraint_settings)
+
+    # Find cut locations
+    opt_out = optimizer.optimize()
+
+    wire_cut_actions = []
+    gate_ids = []
+
+    opt_out = cast(DisjointSubcircuitsState, opt_out)
+    opt_out.actions = cast(list, opt_out.actions)
+    for action in opt_out.actions:
+        if action[0].get_name() == "CutTwoQubitGate":
+            gate_ids.append(action[1][0])
+        else:
+            # We only support four types of optimizer cuts
```

Comment:
@Ibrahim-Shehzad , I updated this commend when I added CutBothWires to the assert statement.

I normally just drop a comment above an assert statement explaining why I am making an assumption about a data structure. The code after this assert makes assumptions about how the Action is formatted, as we are assuming it's one of these 3 classes.

I like asserts for things like this because if we add a new class, instead of a strange index error, this assert would immediately flag our newly broken assumption.

Reply:
Right, that makes sense. I think I missed the update. I guess part of my concern was also whether ''types'' of optimizer cuts could be confused with cuts being "LO" vs "LOCC" but I think what you have is probably just fine. Thanks.

Reply:
I will rephrase/improve this comment

Reply:
The updated comment looks good, thanks!